src/status_im/common/enter_seed_phrase/view.cljs

```diff
@@ -149,21 +139,34 @@
           error-in-words?          (or (not last-partial-word-valid?)
                                        (not butlast-words-valid?))
           upper-case?              (boolean (re-find #"[A-Z]" @seed-phrase))
+          error                    (rf/sub [:enter-seed-phrase/error])
+          on-change-seed-phrase    (fn [new-phrase]
+                                     (when error
+                                       (rf/dispatch [:enter-seed-phrase/clear-error]))
+                                     (when @invalid-seed-phrase?
+                                       (reset! invalid-seed-phrase? false))
+                                     (when @incorrect-seed-phrase?
+                                       (reset! incorrect-seed-phrase? false))
+                                     (reset! seed-phrase new-phrase))
           suggestions-state        (cond
-                                     (or error-in-words?
+                                     (or error
+                                         error-in-words?
                                          words-exceeded?
                                          @invalid-seed-phrase?
                                          @incorrect-seed-phrase?)         :error
                                      (string/blank? @seed-phrase)         :info
                                      (string/ends-with? @seed-phrase " ") :empty
                                      :else                                :words)
-          suggestions-text         (cond
-                                     upper-case?             (i18n/label :t/seed-phrase-words-uppercase)
-                                     words-exceeded?         (i18n/label :t/seed-phrase-words-exceeded)
-                                     error-in-words?         (i18n/label :t/seed-phrase-error)
-                                     @invalid-seed-phrase?   (i18n/label :t/seed-phrase-invalid)
-                                     @incorrect-seed-phrase? (i18n/label :t/seed-phrase-incorrect)
-                                     :else                   (i18n/label :t/seed-phrase-info))
+          suggestions-text         (or error
```

Comment:
Reads a little strange to see `cond` nested inside `or` because the `cond` macro is also meant to unnest conditionals if possible. Just a nitpick, but I would make the first condition:

```clojure
(cond
  error       error
  upper-case? ...
  ...)
```

Reply:
Thank you, fixed